### DISCUSSION TITLE
Dynamically modify authorisation parameters at runtime

### DISCUSSION THREAD

**Discussion by scottlet** [nextauthjs/next-auth#3596] [Help]
I have a website for which I'm trying to integrate next-auth. All done so far, but I'm a bit stuck on the final hurdle.

This website serves both .uk and .ie markets and the authentication method is the same for both (OpenID), but the servers are on different urls for data protection reasons.

I have a header that tells me which country a user is in and using the old client I was easily able to swap out the authorisation redirect_uri parameter using this.

I can't, however, see a nice way to do it using next auth.

Right now I'm doing something like:

```typescript
//[...nextauth].ts
export default async (req, res) => {
  const isIE = req?.headers['country-code'] === 'IE' ? false : true;
  
  const redirectUrl = `${isIE ? authRedirectUrlIre : authRedirectUrl}`;

  const authPages = NextAuth({...[object including redirectUrl]});

  return authPages(req, res)
}
```

But this seems incredibly inefficient. Is there a better way to cover this use case? I can't see anything obvious from the documentation. 

The only other way I could think of to do this would be to have two providers listed separately with different IDs and callback parameters and try and do it that way?

**Comment by balazsorban44** [comment]
This is totally possible, see https://next-auth.js.org/configuration/initialization#advanced-initialization

You were almost there.

**Comment by dileepainivossl** [comment]
> This is totally possible, see https://next-auth.js.org/configuration/initialization#advanced-initialization
> 
> You were almost there.

Any idea how to implement the dynamic scops in next-auth Azure  Provider like this

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/resources-and-scopes.md#dynamic-scopes-and-incremental-consent
